cdap-app-fabric/src/main/java/co/cask/cdap/test/internal/AppFabricClient.java

```diff
@@ -303,4 +312,54 @@ public Location deployApplication(Id.Namespace namespace, String appName,
     return deployedJar;
   }
 
+  public void deployTemplate(Id.Namespace namespace, Id.ApplicationTemplate templateId) {
+    DefaultHttpRequest request = new DefaultHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.PUT,
+      String.format("/v3/namespaces/%s/templates/%s", namespace.getId(), templateId.getId()));
+    MockResponder responder = new MockResponder();
+    adapterHttpHandler.deployTemplate(request, responder, namespace.getId(), templateId.getId());
+    verifyResponse(HttpResponseStatus.valueOf(200), responder.getStatus(), "Failed to deploy template.");
+  }
+
+  public void createAdapter(Id.Adapter id, AdapterConfig config) {
+    DefaultHttpRequest request = new DefaultHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.PUT,
+      String.format("/v3/namespaces/%s/adapters/%s", id.getNamespaceId(), id.getId()));
```

Comment:
yep, i will change them to use the version constant we have, as well as the other hardcoded v3s in this class.